### PR TITLE
[TDL-20774] Integration tests

### PR DESCRIPTION
**Pull request by shantanu73**
# Description of change

Added new integration test suite

- Discovery Test
- All Fields Test
- Automatic Fields Test
- Start Date Test
- Bookmark Test
- Sync Canary Test
- Interrupted Sync Test

# Manual QA steps
 - Ran all integration tests locally
 
# Risks
 - Low risk
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
+import os
+from datetime import datetime as dt
+from datetime import timedelta
+
+from tap_tester import connections, menagerie, runner, LOGGER
+from tap_tester.base_suite_tests.base_case import BaseCase
```
Comment (sgandhi1311) [inline tests/base.py:6]: remove unused imports

File: tests/test_bookmark.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                try:
+                    # Tap expects date format "%Y-%m-%dT%H:%M:%S.000Z" which is not handled by
+                    # base test suite so handling it in the test class explicitly
+                    new_bookmarks[self.get_stream_id(stream)] = {
+                        replication_key:
+                        self.timedelta_formatted(self.parse_date(replication_values[-2]),
+                                                 date_format=self.bookmark_format).replace(".000000Z", ".000Z")}
+                except IndexError as e:
+                    raise Exception(f"{stream}: {replication_values}") from e
+        return new_bookmarks
```
Comment (HarrisonMarcRose) [inline tests/test_bookmark.py:83]: The only thing I see different in this method that was overwritten is the `.replace(".000000Z", ".000Z")}` in the last line.  I think you could make it more visible what you are changing from the original method and don't need to repeat all of the logic.  The change for the try catch should be in the tap-tester method if you think it an improvement.

Can you rewrite this something like:

```
    def calculate_new_bookmarks(self):

    new_bookmarks = BookmarkTest.calculate_new_bookmarks()
    new_bookmarks =  {key: {k, v.replace(".000000Z", ".000Z") for val in value} for key, value in new_bookmarks}
```

I'm not 100% sure of that code but something like that

**Review by HarrisonMarcRose — COMMENTED** [review]
Looks good for the most part.

Not sure why there are missing fields in the all fields test.  We should try to get all fields if we can though I guess there are times where we can't simulate some data.

File: tests/test_bookmark.py (hunk 2 above)
Comment (RushiT0122) [reply tests/test_bookmark.py]: Fixed

**Review by HarrisonMarcRose — APPROVED** [review]

**Review by sgandhi1311 — APPROVED** [review]

**Pull request merged**